midolman/src/main/scala/org/midonet/midolman/DatapathController.scala

```diff
@@ -384,11 +384,11 @@ class DatapathController @Inject() (val driver: DatapathStateDriver,
               if InetAddress.getByAddress(zone._2.toBytes) == inetAddress
         } {
             // Keep the minimum of those accounting for the tunnel overhead
-            val tunnelMtu = (intf.getMtu - overhead).toShort
+            val tunnelMtu = intf.getMtu - overhead
             minTunnelMtu = minTunnelMtu.min(tunnelMtu)
         }
 
-        if (minTunnelMtu == Short.MaxValue) {
+        if (minTunnelMtu == Int.MaxValue) {
             if (minMtu != defaultMtu) {
                 log.info(
                     s"There is no interface in any tunnel zone. Updating " +
```

Comment:
A few lines lower, maybe the bit in the original patch that limits `minTunnelMtu` to be at most `0xffff` should be included as well?

Reply:
Yeah it's a good thing, thanks for pointing that out!